### PR TITLE
addresses filter result errors

### PR DESCRIPTION
**Pull request by mdrillin**
Address issue #290 
- FilterConfigComponent now clears any errors when filters cleared or applied
- changes in ApplicationErrorPage to i18n some text, better display of error details

### CODE REVIEW

**Comment by mdrillin** [comment]
@indraraj - after working on this, I'm thinking we could potentially eliminate usage of WithLoader in FilterTreeComponent?  Then we wouldn't display that whole ApplicationErrorPage when we get an api error here.  wdyt?

File: ui/packages/ui/src/app/components/ApplicationErrorPage.tsx [hunk 1]
```diff
@@ -8,60 +8,72 @@ import {
...
 import { ExclamationTriangleIcon } from "@patternfly/react-icons";
-import { useState } from "react";
 import * as React from "react";
+import { useState } from "react";
+import { useTranslation } from 'react-i18next';
 import "./ApplicationErrorPage.css";
 
 export interface IApplicationErrorPageProps {
   icon?: React.ComponentType<any>;
-  title?: string;
```
Comment (indraraj) [inline ui/packages/ui/src/app/components/ApplicationErrorPage.tsx:20]: The idea to have these optional props was to make this component configurable and use in diff error causing scenarios, whereas the default case will be used in `WithErrorBoundary` component

File: ui/packages/ui/src/app/components/ApplicationErrorPage.tsx [hunk 2]
```diff
@@ -8,60 +8,72 @@ import {
...
+
   return (
     <React.Fragment>
       <PageSection className="ps_error" variant={PageSectionVariants.light}>
         <div className="application-error-page">
           <EmptyState variant={EmptyStateVariant.large}>
             <EmptyStateIcon icon={props.icon || ExclamationTriangleIcon} />
             <Title headingLevel="h5" size="lg">
-              {props.title || "Something is wrong"}
+              {t('appErrorTitle')}
```
Comment (indraraj) [inline ui/packages/ui/src/app/components/ApplicationErrorPage.tsx:52]: This will show server error on all the scenarios even in case of UI breaking.

**Comment by indraraj** [comment]
> @indraraj - after working on this, I'm thinking we could potentially eliminate usage of WithLoader in FilterTreeComponent? Then we wouldn't display that whole ApplicationErrorPage when we get an api error here. wdyt?

Sorry, I'm afraid I am not able to follow you here on why to remove `WithLoader`, and if we want to display something else on API error we can do it  by passing custom `errorChildren` e.g ` errorChildren={<ApiError error={props.errorMsg} />}`

**Comment by mdrillin** [comment]
Yeah true on the WithLoader comment - better alternate to use custom errorChildren.  I can address the couple comments and update

**Comment by mdrillin** [comment]
Updated the PR to address comments

**Review by indraraj — APPROVED** [review]
Tested locally, looks nice.

**Pull request merged**